src/libdrakvuf/libdrakvuf.h

```diff
@@ -404,6 +404,26 @@ char* drakvuf_get_filename_from_handle( drakvuf_t drakvuf,
                                         drakvuf_trap_info_t* info,
                                         addr_t handle );
 
+int drakvuf_event_fd_add(drakvuf_t drakvuf,
+                         int fd,
+                         void (*plugin_cb) (int fd, void* data),
+                         void* data,
+                         int flags);
+
+int drakvuf_event_fd_remove(drakvuf_t drakvuf,
+                            int fd);
+
+/*---------------------------------------------------------
+ * Event FD flags
+ */
+
+typedef enum
+{
+    __INVALID_EVENT_FD,
```

Comment:
Do we really need this enum? I don't see much use for it as we could just create a function that wraps the vmi_events_listen(0) check and then the poll loop check could just do `fd_info->plugin_cb(fd_info->fd, fd_info->data);` for everything and wouldn't need a case-by-case handling

Reply:
Nice idea.  I like that better.  Will work on it.

Reply:
Done.  What do you think?